### PR TITLE
[tiff] Update to 4.5.1

### PR DESCRIPTION
**Pull request by dg0yt**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/tiff/portfile.cmake [hunk 1]
```diff
@@ -31,6 +29,7 @@ vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}"
     OPTIONS
         ${FEATURE_OPTIONS}
+        -DCMAKE_FIND_PACKAGE_PREFER_CONFIG=ON
```
Comment (dg0yt) [inline ports/tiff/portfile.cmake:32]: For liblzma and WebP.
Cf. https://gitlab.com/libtiff/libtiff/-/merge_requests/494#note_1431351045.

File: ports/tiff/portfile.cmake [hunk 2]
```diff
@@ -50,7 +49,10 @@ vcpkg_cmake_configure(
 
 vcpkg_cmake_install()
 
-vcpkg_cmake_config_fixup(CONFIG_PATH "lib/cmake/${PORT}")
+# CMake config wasn't packaged in the past and is not yet usable now,
+# cf. https://gitlab.com/libtiff/libtiff/-/merge_requests/496
+# vcpkg_cmake_config_fixup(CONFIG_PATH "lib/cmake/tiff")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/lib/cmake" "${CURRENT_PACKAGES_DIR}/debug/lib/cmake")
```
Comment (dg0yt) [inline ports/tiff/portfile.cmake:55]: I mean it: don't ship yet.

File: ports/tiff/portfile.cmake [hunk 3]
```diff
@@ -61,7 +63,6 @@ vcpkg_fixup_pkgconfig()
 file(REMOVE_RECURSE
     "${CURRENT_PACKAGES_DIR}/debug/include"
     "${CURRENT_PACKAGES_DIR}/debug/share"
-    "${CURRENT_PACKAGES_DIR}/share"
```
Comment (dg0yt) [inline ports/tiff/portfile.cmake:64]: This is what would have removed any CMake config.

File: ports/tiff/portfile.cmake (hunk 2 above)
Comment (dg0yt) [reply ports/tiff/portfile.cmake:55]: https://gitlab.com/libtiff/libtiff/-/merge_requests/506/diffs

**Comment by dg0yt** [comment]
Basically. this is a straight patch version update. Ping for review @FrankXie05 @BillyONeal.

**Review by FrankXie05 — APPROVED** [review]

**Pull request merged**